### PR TITLE
add antialiasing to FlxTilemap

### PR DESCRIPTION
**Pull request by larsiusprime**

### CODE REVIEW

File: flixel/tile/FlxTilemap.hx [hunk 1]
```diff
@@ -1013,6 +1018,11 @@ class FlxTilemap extends FlxBaseTilemap<FlxTile>
 	}
 	#end
 	
+	private function set_antialiasing(value:Bool):Bool
```
Comment (Gama11) [inline flixel/tile/FlxTilemap.hx]: Why do we need a setter here?

**Comment by Gama11** [comment]
Does this work with both `FlxG.renderTile` and `FlxG.renderBlit`?

File: flixel/tile/FlxTilemap.hx (hunk 1 above)
Comment (larsiusprime) [inline flixel/tile/FlxTilemap.hx]: I'm just following the convention from FlxSprite

**Comment by larsiusprime** [comment]
@Gama11: It works with rendertile, not sure about renderblit. Let me see.

**Comment by larsiusprime** [comment]
Okay, looks like it doesn't -- in BLIT mode, the draw call happens on the FlxTilemapBuffer, which passes a null (and thus false) value for "smoothing" to draw().

Serendipitously, this creates a non-speculative use for the antialiasing setter on the tilemap -- we need to add an antialiasing flag to the FlxTilemapBuffers, and that's a natural place to update it.

**Comment by Gama11** [comment]
I was about to tell you that there's a reason why `FlxSprite` has an empty setter, namely that it's overriden by `FlxSpriteGroup`. Actually looking into that though, that doesn't seem to be the case...

**Comment by Gama11** [comment]
Hm, looks like it was needed by the old version of the bitmap text class, but apparently not anymore in the refactored `FlxBitmapText`: https://github.com/HaxeFlixel/flixel/commit/f4829a8f8f68d1ba93f8d866748cc3a106d76963

**Comment by larsiusprime** [comment]
Okay, I have the change for Blit, tested and works. one second.

File: flixel/tile/FlxTilemap.hx [hunk 2]
```diff
@@ -970,6 +975,10 @@ class FlxTilemap extends FlxBaseTilemap<FlxTile>
 	{
 		var buffer = new FlxTilemapBuffer(_tileWidth, _tileHeight, widthInTiles, heightInTiles, camera, scale.x, scale.y);
 		buffer.pixelPerfectRender = pixelPerfectRender;
+		if (FlxG.renderBlit)
```
Comment (Beeblerox) [inline flixel/tile/FlxTilemap.hx]: you can skip this check for render mode and just set `antialiasing` directly

File: flixel/tile/FlxTilemap.hx [hunk 3]
```diff
@@ -1013,6 +1022,19 @@ class FlxTilemap extends FlxBaseTilemap<FlxTile>
 	}
 	#end
 	
+	private function set_antialiasing(value:Bool):Bool
+	{
+		antialiasing = value;
+		if (FlxG.renderBlit)
```
Comment (Gama11) [inline flixel/tile/FlxTilemap.hx]: Same with this check I guess.

**Pull request merged**